File: JavaPools/PoolsLibrary/src/main/java/ru/urfu/javapools/poolslibrary/objectutilization/IObjectUtilizer.java
package ru.urfu.javapools.poolslibrary.objectutilization;

public interface IObjectUtilizer<TK,TV> {
	
	void utilize(TK key, TV pooObject, Object caller);
	
	void addListener(IObjectUtilizationListener<TK,TV> listener);
	void removeListener(IObjectUtilizationListener<TK,TV> listener);	
	int getListenersCount();
}

File: CSharpPools/PoolsLibrary/ObjectUtilization/IObjectUtilizer.cs
using System;

namespace PoolsLibrary.ObjectUtilization
{
    public interface IObjectUtilizer<TK,TV>
    {
        event EventHandler<GoneObjectEventArgs<TK,TV>> ObjectIsGone;

        void Utilize(TK key, TV pooObject, object caller);
    }
}